### ISSUE TITLE
Should we exclude or include pyproject.toml

### ISSUE THREAD

**Issue by kbakk** [camunda-community-hub/pyzeebe#183]
Each time I install using `pipenv install --dev` I get a `pyproject.toml` generated in the project:

```toml
[build-system]
requires = ["setuptools", "wheel"]
build-backend = "setuptools.build_meta:__legacy__"
```

I don't really understand where this is coming from. It seems to be setuptools related. It's only when running with `--dev` flag it's created. 

Should it be included or ignored?

Somewhat on the same topic, I think this project could benefit from changing from setup.py to pyproject.toml and flit/poetry. flit would only manage the packaging aspect of this project, while with poetry it would do that in addition to managing virtualenvironments and dev dependencies (i.e. replace pipenv).

**Comment by JonatanMartens** [comment]
I have been wanting to switch to poetry for a while now. 

Once `pre-release/3.0.0` is merged into `development` (soon I hope) and we bump dependencies (see #174  for example). I'll open a pr once these are done.

> flit would only manage the packaging aspect of this project

Is there a reason not to use `poetry publish` instead? I'm not really familiar with `flit`.

**Comment by kbakk** [comment]
flit is only for packaging, so it would either be flit+pipenv, or poetry alone. I am using poetry for all work projects, and am super happy with it - can recommend it (though, it's not entirely without its problems, but that is more because of the python packaging ecosystem). 

I have only used a private PyPI, which came with its own publish cli that was convenient to use, so haven't used poetry for publishing - but I think `poetry publish` should do the job just fine.
